README.md

```diff
@@ -1,44 +1,86 @@
 # Ř
 
-To give it a spin try
+## Giving it as spin
+
+The easiest way to try Ř is using our pre-built docker container
 
     docker run -it registry.gitlab.com/rirvm/rir_mirror:master /opt/rir/build/release/bin/R
 
-## Building from Source
+## Building
+
+If you want to build Ř from source, we strongly recommend that you use Ubuntu.
+This way, the build scripts can download pre-compiled LLVM binaries instead of compiling LLVM from scratch.
+If you're not sure if your OS can benefit from pre-compiled LLVM binaries, check the tools/fetch-llvm.sh script.
+
+Before we can begin, we must install the dependencies.
+The optional ninja-build dependency improves the compilation time.
+For the R build-dep step you may need to enable source code repositories (deb-src) via GNOME Software or /etc/apt/sources.list.
 
-This is currently only tested on Ubuntu.
+    sudo apt install build-essential cmake curl
+    sudo apt install ninja-build
+    sudo apt build-dep r-base
 
+Then, we can proceed with the compilation:
+
+    # Clone this repository
     git clone https://github.com/reactorlabs/rir
     cd rir
-    cmake -DCMAKE_BUILD_TYPE=build_type .    # possible build_types: release, debugopt, debug
-    # Fetch and build dependencies. This will build gnur from source, which takes a while.
-    # To build llvm from source (for debug symbols) set the following. This can take a long time to build.
-    # export BUILD_LLVM_FROM_SRC=1
-    make setup
-    # Run cmake again, this should now say "Found LLVM 11"
-    cmake .
-    make
 
-Now run Ř with
+    # Create a directory for the build.
+    # You can use any name.
+    mkdir build
+    cd build
+
+    # Invoke cmake for the first time.
+    # Possible build types: release, debugopt, debug
+    cmake -GNinja -DCMAKE_BUILD_TYPE=release -DFORCE_COLORED_OUTPUT=1 ..
+
+    # Fetch and/or build LLVM and Gnu R.
+    # Ubuntu and MacOS download pre-compiled LLVM binaries, which takes around 10 minutes.
+    # On other systems, or if you set BUILD_LLVM_FROM_SRC, this takes a very long time.
+    ninja setup
+
+    # Now we can build Ř itself.
+    # The cmake should output "Found LLVM 12"
+    cmake ..
+    ninja
+
+Congratulations! You can now run Ř with
 
     bin/R
 
-If you want to develop Ř then it is highly recommended to use off-tree builds and ninja instead of make. See further down (Hacking).
+### In-tree builds
```

Comment:
drop? is there any need for that?

Reply:
Sounds reasonable, YAGNI. I suppose that if someone really wants in-tree builds, they are free to figure out the cmake commands themselves.